ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -265,6 +231,40 @@ public Mode suspendScan (boolean async)
    {
        synchronized (_stateLock)
        {
+           // see if suspension should delay until certain types have recovered
+           if (recoveryPropertyManager.getRecoveryEnvironmentBean().isWaitForFinalRecovery()) {
+               // List<String> types = recoveryPropertyManager.getRecoveryEnvironmentBean().getTypeNamesToBlockShutdown();
+
+               // disable the transaction system but keep recovery running
+               TxControl.disable();
+               // wait for in flight transactions to complete
+               TransactionReaper.terminate(true);
+
+               while (true) {
+                   try {
+                       // Need to be sure that a scan has fully completed, for now illustrate this by waiting to be used outside of scanning
+                       // maybe it can be checked we are not in the middle of scanning rather than simply wait?
+                       _stateLock.wait();
+                       // preventShutdown is reset in doWorkInternal to false and after scanning completes check if there were any reasons to preventShutdown
+                       // I don't know the best place to make this comment but checking for subordinates could be a new recovery module wrapping something like https://github.com/jbosstm/narayana/blob/86182416bea64368ecdfc7e78767f798b15c14db/ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java#L438C4-L438C4
+                       // but please know that these checks are called by external processes at any time and so that would need to be blocked while this suspension process continues and appropriately handled
+                       if (!blockSuspension) {
+                           // if (!storeContainsAnyOf(types) && !hasOrphans() && !hasSubordinates()) {
+                           // the store does not contain any records we're interested in,
+                           // so move directly to Mode.TERMINATED
```

Comment:
@jmfinelli this comment is now out of date as it implies we would move to Mode.TERMINATED but it would be Mode.SUSPENDED

Reply:
That being said, I am thinking that the way it was in shutdown section and knowing we would go to terminate might have given some assumptions about the possible future actions that aren't applicable in suspend. Do we definitely know that no further requests for scanning can take place until it is is resumed?

Reply:
AFAIK, yep, PeriodicRecovery shouldn't be able to run further scan requests while it is SUSPENDED.